OpenWeather/Data/Adapters/LocationAdapter.swift

```
@@ -7,12 +7,39 @@
 
 import Foundation
 import CoreLocation
+import GRDB
+
+enum LocationAdapter {
+    static func device(from: LocationData) -> DeviceLocation {
+        DeviceLocation(
+            name: from.name,
+            latitude: from.lat,
+            longitude: from.long
+        )
+    }
+
+    static func device(from: Row) throws -> DeviceLocation? {
+        guard
+            let name = from[0] as? String,
+            let lat = from[1] as? Double,
+            let long = from[2] as? Double
+        else {
+            // TODO: @0xLeif, should we throw an error instead?
```

Yes! We should throw an error if the Row data isn't as expected.